distribution/lib/Standard/Base/0.0.0-dev/src/Data/Json.enso

```diff
@@ -281,3 +102,149 @@ type Marshalling_Error
         Marshalling_Error.Missing_Field _ field _ ->
             "Missing field in Json: the field `" + field.to_text "` was missing in the json."
+
+type JS_Object
+    ## Creates a JS_Object from a list of key-value pairs.
+       Keys must be `Text` values.
+       Values will be recursively converted to JSON serializable as needed.
+    from_pairs : Vector -> JS_Object
+    from_pairs pairs =
+        js_object = pairs.fold new_object current->pair->
+            case pair.at 0 of
+                text : Text ->
+                    set_value current text (pair.at 1).to_js_object
+                    current
+                _ -> Error.throw (Illegal_Argument.Error "JS_Object.from_pairs: key must be a Text value")
+        JS_Object.Value js_object
+
+    ## PRIVATE
+    Value js_object
+
+    ## Get a value for a key of the object, or a default value if that key is not present.
+
+       Arguments:
+       - key: The key to get.
+       - if_missing: The value to return if the key is not found.
+    get : Text -> Any -> JS_Object | Boolean | Number | Nothing | Text | Vector
+    get self key ~if_missing=Nothing =
+        if (has_property self.js_object key) . not then if_missing else
+            value = get_value self.js_object key
+            make_enso value
+
+    ## Get a value for a key of the object.
+       If the key is not found, throws a `No_Such_Key` error.
+
+       Arguments:
+       - key: The key to get.
+    at : Text -> JS_Object | Boolean | Number | Nothing | Text | Vector
+    at self key = self.get key (Error.throw (No_Such_Key.Error self key))
+
+    ## Get the keys of the object
+    field_names : Vector
+    field_names self =
+        Vector.from_polyglot_array (get_property_names self.js_object)
+
+    ## Gets the number of keys in the object
+    length : Number
+    length self =
+        get_property_names self.js_object . length
+
+    ## Convert the object to a Vector of Pairs
+    to_vector : Vector
+    to_vector self =
+        keys = get_property_names self.js_object
+        proxy = Array_Proxy.new keys.length (i-> Pair.new (keys.at i) (self.get (keys.at i)))
+        Vector.from_polyglot_array proxy
+
+    ## Convert the object to a JSON
+    to_js_object : JS_Object
+    to_js_object self = self
+
+    ## Convert to string
+    to_text : Text
+    to_text self = Json.stringify self
+
+    ## Checks if this JS_Object is equal to another JS_Object.
+
+       Arguments:
+       - that: The map to compare `self` to.
+    == : JS_Object -> Boolean
+    == self that = case that of
+        _ : JS_Object ->
+            self_keys = self.field_names
+            that_keys = that.field_names
+            self_keys.length == that_keys.length && self_keys.all key->
+                (self.get key == that.at key).catch No_Such_Key.Error _->False
+        _ -> False
+
+    ## Render the object for display
+    render : Integer -> Integer -> Text
+    render self depth=0 max_depth=5 = render self depth max_depth
```

Comment:
We have `to_text`, `to_display_text` and now `render`.

This is getting complicated. Not necessarily now, but I think we maybe should revise our to-text-API?

What is the purpose of `render`? How does it differ (if at all) from the intent behind `to_display_text`?

Reply:
Ok, looking further I see that `render` is actually used as backing for `to_default_visualization_data`. Sorry, it's hard to connect the dots from the start in a big PR

Reply:
Is just `to_default_visualization_data` so have merged them.